extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -0,0 +1,389 @@
+        'should have "data-dismiss-href" attribute.');
+    /** @private @const {string} */
+    this.dismissHref_ = assertHttpsUrl(
+        this.element.getAttribute('data-dismiss-href'), this.element);
+
+    /** @private {?function(boolean)} */
+    this.dialogResolve_ = null;
+
+    /** @private {?Promise<string>} */
+    this.dialogPromise_ = null;
+
+    this.userNotificationManager_
+        .registerUserNotification(this.elementId_, this);
+
+    this.registerAction('dismiss', this.dismiss.bind(this));
+  }
+
+  /**
+   * @override
+   */
+  getId() {
+    return this.elementId_;
+  }
+
+  /**
+   * Executes a `POST` request to the url given on the `data-show-if-href`
+   * attribute.
+   * @param {string} ampUserId
+   * @return {!Promise<!PostResponseMetadataDef>}
+   * @private
+   */
+  postShowEndpoint_(ampUserId) {
+    // TODO: Determine wether this should be a POST with json body
+    // or a GET with data on the query params.
+    // see: https://github.com/ampproject/amphtml/issues/1228
+    this.ampUserId_ = ampUserId;
+    const getReq = {
+      method: 'POST',
+      credentials: 'include',
+      /** @type {PostRequestMetadataDef} */
+      body: {
+        'elementId': this.elementId_,
+        'ampUserId': ampUserId
+      }
+    };
+
+    return xhrFor(this.win_).fetchJson(this.showIfHref_, getReq);
+  }
+
+  /**
+   * Creates an POST to the specified `data-dismiss-href` url.
+   * @private
+   * @return {!Promise}
+   */
+  postDismissEnpoint_() {
+    return xhrFor(this.win_).fetchJson(this.dismissHref_, {
+      method: 'POST',
+      credentials: 'include',
+      body: {
+        'elementId': this.elementId_,
+        'ampUserId': this.ampUserId_
+      }
+    });
+  }
+
+  /**
+   * Success handler for `postShowEndpoint_`.
+   * @param {!PostResponseMetadataDef}
+   * @return {!Promise<boolean>}
+   * @private
+   */
+  onPostShowEndpointSuccess_(data) {
+    assert(typeof data['showNotification'] == 'boolean', '`showNotification` ' +
+        'should be a boolean. Got "%s" which is of type %s.',
+        data['showNotification'], typeof data['showNotification']);
+
+    if (data['showNotification']) {
+      return Promise.resolve(data['showNotification']);
+    }
+    // if no notification is shown, resolve right away with `false` since
+    // we dont need to show it.
+    return this.dialogResolve_(false);
+  }
+
+  /**
+   * Get async cid service.
+   * @return {!Promise}
+   * @private
+   */
+  getAsyncCid_() {
+    this.dialogPromise_ = new Promise((resolve, reject) => {
+      this.dialogResolve_ = resolve;
+    });
+    return cidFor(this.win_).then(cid => {
+      // `amp-user-notification` is our cid scope, while we give it a resolved
+      // promise for the 2nd argument so that the 3rd argument (the
+      // persistentConsent) is the one used to resolve getting
+      // the external CID.
+      return cid.get('amp-user-notification',
+          Promise.resolve(), this.dialogPromise_);
```

Comment:
currently, from what me and @cramforce  have discussed. it shouldnt be rejected. the user only really has 1 option to accept/dismiss (to resolve) the notification or have the nagging notification sitting there (to never resolve).